### PR TITLE
[WIP] Introduce more intuitive method names in Data classes and reduce required input parameter for TrajectoryData.set_trajectory() method

### PR DESCRIPTION
**Pull request by astamminger**
The changes in this pull request address issue #201 providing more intuitive method names. I clustered the changes into three major topics and would be glad to hear your opinion

**1. Renamed functions**
* ``ArrayData``
  - ``.iterarrays()`` -> ``.get_iterarrays()``
* ``TrajectoryData``
  -  ``._get_cif()`` -> ``.get_cif()``
  - ``._get_aiida_structure`` -> ``.get_structure()``
* ``StructureData``
  - ``._get_cif()`` -> ``.get_cif()``
* ``Code``
  - ``.full_text_info()`` -> ``.get_full_text_info()``

Functions have been renamed and the tests have been adopted accordingly. (Deprecation warnings have been added to all "old" functions)

**2. Methods changed to properties**
* ``Code``
  - ``.is_hidden()`` -> ``.hidden``
* ``RemoteData``
  - ``.is_empty()`` -> ``.is_empty``
* ``StructureData``, ``Kind``
  - ``.is_alloy()`` -> ``.is_alloy``
  - ``.has_vacancies()`` -> ``.has_vacancies``

Idea was that these values are rather "attributes" of these classes and it is more intuitive to treat them as properties rather than actual methods. Question here is: Is this OK and how to exactly deal with this since we cannot simply add deprecation warnings.

**3. Changes in ``TrajectoryData.set_trajectory()``:**
Input parameters ``stepid`` and ``cells`` are now optional and the implemented behavior is as follows: 
* If ``cells`` parameter is missing: Nothing will be stored for cells. 
* If ``stepids`` parameter is missing: ``stepids`` defaults to the consecutive sequence [0,1,...,len(positions)-1]

Element symbols passed to the ``set_trajectory()`` method are now passed as a list (instead of a numpy.ndarray) and are stored in the ``TrajectoryData.symbols`` attribute (as requested in #201 to simplify querying)

### CODE REVIEW

File: aiida/orm/data/array/trajectory.py [hunk 1]
```diff
@@ -50,30 +50,36 @@ def _internal_validate(self, stepids, cells, symbols, positions, times, velociti
...
+            numsteps = stepids.size
+            if stepids.shape != (numsteps,):
+                raise ValueError("TrajectoryData.stepids must be a 1d array")
+        else:
+            numsteps = positions.shape[0]
+        if cells is not None:
+            if cells.shape != (numsteps, 3, 3):
+                raise ValueError("TrajectoryData.cells must have shape (s,3,3), " "with s=number of steps")
+        numatoms = len(symbols)
+        if numpy.array(symbols).shape != (numatoms,):
```
Comment (giovannipizzi) [inline aiida/orm/data/array/trajectory.py]: Probably we don't need to go via a numpy array just to get the shape

File: aiida/orm/data/array/trajectory.py [hunk 2]
```diff
@@ -135,11 +146,25 @@ def set_trajectory(self, stepids, cells, symbols, positions, times=None, velocit
...
         .. todo :: Choose suitable units for velocities
         """
+
+        import numpy
+
         self._internal_validate(stepids, cells, symbols, positions, times, velocities)
-        self.set_array('steps', stepids)
-        self.set_array('cells', cells)
-        self.set_array('symbols', symbols)
+        # set symbols as attribute for easier querying
```
Comment (giovannipizzi) [inline aiida/orm/data/array/trajectory.py]: Here we can do `list(symbols)` to make sure it's a list?

File: aiida/orm/data/array/trajectory.py [hunk 3]
```diff
@@ -50,30 +50,36 @@ def _internal_validate(self, stepids, cells, symbols, positions, times, velociti
         """
         import numpy
 
-        if not isinstance(stepids, numpy.ndarray) or stepids.dtype != int:
-            raise TypeError("TrajectoryData.stepids must be a numpy array of integers")
-        if not isinstance(cells, numpy.ndarray) or cells.dtype != float:
-            raise TypeError("TrajectoryData.cells must be a numpy array of floats")
-        if not isinstance(symbols, numpy.ndarray):
-            raise TypeError("TrajectoryData.symbols must be a numpy array")
+        if not isinstance(symbols, list):
```
[the remark this answers is not in the document]
Comment (astamminger) [reply aiida/orm/data/array/trajectory.py]: Updated type-check to use ``collections.Iterable`` in 03ff7a5782bf5eb7547a23c88e11f5c5d3db7055

File: aiida/orm/data/array/trajectory.py (hunk 1 above)
Comment (astamminger) [reply aiida/orm/data/array/trajectory.py]: Adjusted in 467d37b42022f6248bc7056bf6f55992a2a7fce6. I removed this test completely b/c there is already this check:
https://github.com/aiidateam/aiida_core/blob/467d37b42022f6248bc7056bf6f55992a2a7fce6/aiida/orm/data/array/trajectory.py#L56-L57
which would trigger immediately if if the list is not 1D. So, this test seems to be superfluous b/c we actually never reach it.

File: aiida/orm/data/array/trajectory.py (hunk 2 above)
Comment (astamminger) [reply aiida/orm/data/array/trajectory.py]: Yep, implemented that in 016a59d05265924edf68b46db30b79d80a0cae82

**Comment by astamminger** [comment]
Renamed ``is_hidden``-> ``hidden``: 980b34d79fa6c99ed6f3aae817598d52dbfdfbd0
Documented changes (#2320): 44b2e8814068ca2c5a51bc39bbf554b720d9cf5f

**Comment by giovannipizzi** [comment]
Squash-merged into a working branch to keep attribution to @astamminger and we'll continue working from there

**Pull request merged**